platform/api/src/video_event_handler.rs

```diff
@@ -0,0 +1,94 @@
+use std::sync::Arc;
+
+use anyhow::Context;
+use common::database::Ulid;
+use pb::scuffle::video::v1::types::{event, Event};
+use pb::scuffle::video::v1::{EventsAckRequest, EventsFetchRequest};
+use prost::Message;
+
+use crate::global::ApiGlobal;
+use crate::subscription::SubscriptionTopic;
+
+pub async fn run<G: ApiGlobal>(global: Arc<G>) -> anyhow::Result<()> {
+	loop {
+		let mut event_stream = global
+			.video_events_client()
+			.clone()
+			.fetch(EventsFetchRequest {
+				target: pb::scuffle::video::v1::events_fetch_request::Target::Room.into(),
+				max_delay_ms: 1000,
+				max_events: 100,
+			})
+			.await
+			.context("failed to fetch events")?
+			.into_inner();
+
+		while let Some(msg) = event_stream.message().await? {
+			if let Some(Event {
+				event: Some(event::Event::Room(event)),
+				timestamp,
+				event_id: Some(evt_id),
+			}) = msg.event
+			{
+				let action = match handle_room_event(&global, event, timestamp).await {
+					Ok(_) => pb::scuffle::video::v1::events_ack_request::Action::Ack(true),
+					Err(err) => {
+						tracing::warn!(err = %err, "failed to handle event, requeueing");
+						pb::scuffle::video::v1::events_ack_request::Action::RequeueDelayMs(5000)
+					}
+				};
+				global
+					.video_events_client()
+					.clone()
+					.ack(EventsAckRequest {
+						id: Some(evt_id),
+						action: Some(action),
+					})
+					.await?;
+			}
+		}
+	}
+}
+
+async fn handle_room_event<G: ApiGlobal>(global: &Arc<G>, event: event::Room, timestamp: i64) -> anyhow::Result<()> {
+	let room_id = event.room_id.as_ref().unwrap();
+	match event.event.context("no event")? {
+		event::room::Event::Ready(..) => {
+			let (channel_id,): (common::database::Ulid,) = sqlx::query_as("UPDATE users SET channel_live_viewer_count = 0, channel_live_viewer_count_updated_at = NOW(), channel_last_live_at = $1 WHERE channel_room_id = $2 RETURNING id")
```

Comment:
Technically the viewer count can be non zero by the time the stream is ready.
If the video transcoder is restarted mid-stream it will emit a new room ready event.

Reply:
You also should keep track of the active_connection_id. 
Its apart of the ready event. This allows us to figure out which disconnect events are still valid.